crypto/secp256k1/secp256k1.go

```diff
@@ -80,8 +81,13 @@ func (privKey PrivKeySecp256k1) Equals(other crypto.PrivKey) bool {
 // It uses OS randomness in conjunction with the current global random seed
 // in tendermint/libs/common to generate the private key.
 func GenPrivKey() PrivKeySecp256k1 {
+	return genPrivKey(crypto.CReader())
+}
+
+// genPrivKey generates a new secp256k1 private key using the provided reader.
+func genPrivKey(reader io.Reader) PrivKeySecp256k1 {
 	privKeyBytes := [32]byte{}
-	copy(privKeyBytes[:], crypto.CRandBytes(32))
+	io.ReadFull(reader, privKeyBytes[:])
```

Comment:
This can return an `error` (and I don't see why we can expect it not to happen here). I think we should panic in that case (e.g. there is not enough entropy available to create a private key -> panic).

Reply:
My bad, forgot that io.readfull errors